### PR TITLE
:sparkles: Visualizing in RViz Tutorial

### PR DESCRIPTION
**Pull request by tylerjw**
This PR is based on:

- #262 

### Description

This new tutorial should introduce new users to MoveItVisualTools very quickly and provide some links for how they can learn more about its features and use cases.

### CODE REVIEW

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 1]
```diff
@@ -0,0 +1,290 @@
+Visualizing In RViz
+===================
+
+This tutorial will introduce you to a tool that can help you quickly understand what your MoveIt program is doing my rendering visualizations in RViz.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:4]:
```suggestion
This tutorial will introduce you to a tool that can help you quickly understand what your MoveIt program is doing by rendering visualizations in RViz.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    // Create the MoveIt MoveGroup Interface
+    ...
+
+    // Shutdown ROS
+    rclcpp::shutdown();  // <--- This will cause the spin function in the thread to return
+    spinner.join();  // <--- Join the thread before exiting
+    return 0;
+  }
+
+After each one of these changes you should rebuild your workspace to make sure you don't have any syntax errors.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:83]:
```suggestion
After each one of these changes, you should rebuild your workspace to make sure you don't have any syntax errors.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+3 Create and Initialize MoveItVisualTools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next we will construct and initialize MoveItVisualTools after the construction of MoveGroupInterface.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:88]:
```suggestion
Next, we will construct and initialize MoveItVisualTools after the construction of MoveGroupInterface.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    auto move_group_interface = MoveGroupInterface(node, "panda_arm");
+
+    // Construct and initialize MoveItVisualTools
+    auto moveit_visual_tools = moveit_visual_tools::MoveItVisualTools{
+        node, "panda_link0", rviz_visual_tools::RVIZ_MARKER_TOPIC,
+        move_group_interface.getRobotModel()};
+    moveit_visual_tools.deleteAllMarkers();
+    moveit_visual_tools.loadRemoteControl();
+
+In the constructor we pass in the ROS node, the base link of the robot, the marker topic to use (more on this later), and the robot model (which we get from the move_group_interface).
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:103]: This sentence reads a bit awkwardly. I think it's missing a comma, but even so, it sounds a bit Yoda-like. Not sure that my suggestion below is the best wording, but I'd re-word it one way or another.

```suggestion
We pass the following into the constructor: the ROS node, the base link of the robot, the marker topic to use (more on this later), and the robot model (which we get from the move_group_interface).
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+Next we make a call to delete all the markers, this clears any rendered state out of RViz that we have left over from previous runs.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:104]:
```suggestion
Next, we make a call to delete all the markers. This clears any rendered state out of RViz that we have left over from previous runs.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+Lastly we load remote control.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:105]:
```suggestion
Lastly, we load remote control.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+Remote control is a really simple plugin that lets us have a button in RViz to interact with our program.
+
+4 Write closures for visualizations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After we've constructed and initialized we'd like to create some closures (function objects that have access to variables in our current scope) that we can use later in our program to render visualizations in RViz.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:111]:
```suggestion
After we've constructed and initialized, we now create some closures (function objects that have access to variables in our current scope) that we can use later in our program to help render visualizations in RViz.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+Each of these call a function on ``moveit_visual_tools`` that changes something in RViz.
+The first one, ``draw_title`` adds text one meter above the base of the robot.
+This is a useful way to show the state of your program from a high level.
+The second one calls a function called ``prompt``.
+This function blocks your program until the user presses the ``next`` button in RViz.
+This is helpful for stepping through a program when debugging.
+The last one draws the tool path of a trajectory that we have planned.
+This is often helpful for understanding a planned trajectory from the perspective of the tool.
+
+You might be asking yourself why we would create lambdas like this and the reason is simply to make the code that comes later easier to read and understand.
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:145]:
```suggestion
You might be asking yourself why we would create lambdas like this, and the reason is simply to make the code that comes later easier to read and understand.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+      msg.orientation.w = 1.0;
+      msg.position.x = 0.28;
+      msg.position.y = -0.2;
+      msg.position.z = 0.5;
+      return msg;
+    }();
+    move_group_interface.setPoseTarget(target_pose);
+
+    // Create a plan to that target pose
+    prompt("Press 'next' in the RvizVisualToolsGui window to plan");
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:169]:
```suggestion
    prompt("Press 'Next' in the RvizVisualToolsGui window to plan");
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+      moveit::planning_interface::MoveGroupInterface::Plan msg;
+      auto const ok = static_cast<bool>(move_group_interface.plan(msg));
+      return std::make_pair(ok, msg);
+    }();
+
+    // Execute the plan
+    if (success) {
+      draw_trajectory_tool_path(plan.trajectory_);
+      moveit_visual_tools.trigger();
+      prompt("Press 'next' in the RvizVisualToolsGui window to execute");
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:182]:
```suggestion
      prompt("Press 'Next' in the RvizVisualToolsGui window to execute");
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+
+  cd ~/ws_moveit2
+  source install/setup.bash
+  ros2 run hello_moveit hello_moveit
+
+You'll notice that your program has stopped with a log that looks like this:
+
+.. code::
+
+  [INFO] [1652822889.492940200] [hello_moveit.remote_control]: Waiting to continue: Press 'next' in the RvizVisualToolsGui window to plan
```
Comment (stephanie-eng) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:264]:
```suggestion
  [INFO] [1652822889.492940200] [hello_moveit.remote_control]: Waiting to continue: Press 'Next' in the RvizVisualToolsGui window to plan
```

**Review by stephanie-eng — COMMENTED** [review]
I still need to follow along with the steps and try this, but fo rnow I've left some comments on spelling/grammar

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+This is often helpful for understanding a planned trajectory from the perspective of the tool.
+
+You might be asking yourself why we would create lambdas like this, and the reason is simply to make the code that comes later easier to read and understand.
+As your write software it is often helpful to break up your functionality into named functions which can be easily reused and tested on their own.
+You will see in the next section how we use these functions we created.
+
+5 Visualize the steps of your program
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here I've augmented the code in the middle of your program with some new code.
```
Comment (abake48) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:152]:
```suggestion
Now we'll augment the code in the middle of your program.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+.. image:: panel_menu.png
+
+Then select ``RvizVisualToolsGui`` and click OK.
+This will create a new panel on the bottom left with a ``Next`` button we'll use later.
+
+.. image:: add_rviz_tools_gui.png
+
+.. image:: next_button.png
+
+Finally we need to add a ``Marker Array`` to render the visualizations we've added.
```
Comment (abake48) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:234]:
```suggestion
Finally, we need to add a ``Marker Array`` to render the visualizations we've added.
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+Scroll to the bottom of the items in the Displays panel and edit the topic that the new Marker Array is using to ``/rviz_visual_tools``.
+
+.. image:: marker_array_topic.png
+
+You are now ready to run your new program with visualizations.
+
+7 Run the Program
+^^^^^^^^^^^^^^^^^
+
+In a new terminal go to the workspace, source the workspace, and run ``hello_moveit``:
```
Comment (abake48) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:252]:
```suggestion
In a new terminal, go to the workspace, source the workspace, and run ``hello_moveit``:
```

**Review by abake48 — CHANGES_REQUESTED** [review]
Working on running through the tutorial right now but I have a few nitpicky grammatical suggestions from my read through.

**Comment by tylerjw** [comment]
> Working on running through the tutorial right now but I have a few nitpicky grammatical suggestions from my read through.

I've accepted your changes, would you mind re-reviewing this?

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 15]
```diff
@@ -0,0 +1,290 @@
...
+1 Add the dependency moveit_visual_tools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add this line to your ``package.xml`` in the ``hello_moveit`` project after the other ``<depend>`` statements:
+
+.. code-block:: xml
+
+  <depend>moveit_visual_tools</depend>
+
+Then in your ``CMakeLists.xml`` add this line to the section of ``find_package`` statements:
```
Comment (abake48) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:24]:
```suggestion
Then in your ``CMakeLists.txt`` add this line to the section of ``find_package`` statements:
```

File: doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst [hunk 16]
```diff
@@ -0,0 +1,290 @@
...
+
+.. code::
+
+  [INFO] [1652822889.492940200] [hello_moveit.remote_control]: Waiting to continue: Press 'Next' in the RvizVisualToolsGui window to plan
+
+Click the ``Next`` button in RViz and see your application advance.
+
+.. image:: planning.png
+
+You'll see after you clicked the next button your application planned, added a title above the robot, and drew a line representing the tool path.
```
Comment (abake48) [inline doc/tutorials/visualizing_in_rviz/visualizing_in_rviz.rst:270]:
```suggestion
You'll see after you clicked the next button, your application planned, added a title above the robot, and drew a line representing the tool path.
```

**Review by abake48 — COMMENTED** [review]
Couple more changes and I'll feel comfortable approving this PR

**Review by abake48 — APPROVED** [review]
lgtm!

**Pull request merged**